src/DataTypes/DataTypesDecimal.h

```diff
@@ -37,8 +37,11 @@ class DataTypeDecimal final : public DataTypeDecimalBase<T>
     using Base::Base;
 
     static constexpr auto family_name = "Decimal";
+    static constexpr auto mysql_name = "DECIMAL";
 
     const char * getFamilyName() const override { return family_name; }
+    const char * getSQLCompatibleName() const override { return mysql_name; }
```

Comment:
It will be cleaner to include `M` and `D` parameters: https://dev.mysql.com/doc/refman/8.0/en/fixed-point-types.html